posts/2020-11-20-JNDI-gRPC.adoc

```diff
@@ -0,0 +1,361 @@
+
+[#grpc]
+== Provide and consume gRPC services from your web applications
+
+link:https://grpc.io/docs/what-is-grpc/introduction/[gRPC] is a high-performance, open source universal RPC framework. gRPC support on Liberty allows developers to both provide and consume gRPC services from your web applications. The introduction of gRPC support in Open Liberty now means you can take advantage of the benefits of gRPC more easily than before. Those benefits include great performance, simple service definitions via Protocol Buffers, cross-platform and language support, and wide industry adoption.
+
+These two features which were previously in our beta are now available in this months GA: `grpc-1.0`, which enables gRPC services, and `grpcClient-1.0`, which enables the use of a gRPC client for outbound calls.
+
+The `grpc-1.0` feature works by scanning web apps for gRPC service implementations, through implementors of `io.grpc.BindableService`. The web app must include the protocol buffer compiler-generated code for the services it intends to provide, and additionally the service class must provide a no-argument constructor. The web app does not need to include any core gRPC libraries; those are provided by the Liberty runtime. Once a gRPC service is scanned and started, it becomes accessible to remote gRPC clients on the configured HTTP ports.
+
+The `grpcClient-1.0` feature provides applications with access to a link:https://netty.io/[Netty] gRPC client, as well as the related libraries. A web app must provide a client implementation and stubs, and can make outbound calls with a `io.grpc.ManagedChannel` without needing to provide the supporting client libraries.
+
+You can now sample gRPC by either following the instructions in this link:https://github.com/OpenLiberty/sample-grpc[GitHub repositories] README or following this basic Hello World service (add the  `grpc-1.0` to the `server.xml`):
+
+[source, java]
+----
+package com.ibm.ws.grpc;
+
+import com.ibm.ws.grpc.beans.GreetingBean;
+
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.stub.StreamObserver;
+
+public class HelloWorldService extends GreeterGrpc.GreeterImplBase {
+
+    public HelloWorldService(){}
+
+    @Override
+    public void sayHello(HelloRequest req, StreamObserver<HelloReply> responseObserver) {
+        HelloReply reply = HelloReply.newBuilder().setMessage("Hello " + req.getName()).build();
+        responseObserver.onNext(reply);
+        responseObserver.onCompleted();
+    }
+}
+----
+
+For this example, the application must provide the link:https://github.com/grpc/grpc-java/blob/master/examples/src/main/proto/helloworld.proto[helloworld protof definition] along with the protobuf compiler output. No additional libraries need to be provided with the application, and once it's started the helloworld greeter service will be accessible on the server's HTTP endpoints.
```

Comment:
@wtlucy should confirm, but I think "protof" should be "protobuf".
```suggestion
For this example, the application must provide the link:https://github.com/grpc/grpc-java/blob/master/examples/src/main/proto/helloworld.proto[helloworld protobuf definition] along with the protobuf compiler output. No additional libraries need to be provided with the application, and once it's started the helloworld greeter service will be accessible on the server's HTTP endpoints.
```

Reply:
Done